juspay/hyperswitch
# refactor(payments_v2): use batch encryption for intent create and confirm intent

## Type of Change



## Description

This PR adds batch encryption for the payment intent create and payment intent confirm apis. This also refactors the `to_encryptable` operation to fix a small bug which considers the inner type and ignores the annotated types in certain conditions.

This PR also adds a domain type for address which will be used across the application instead of the api models ( which is the current scenerio ). This will be used in the subsequent PRs.


## Motivation and Context

In higher environments, the encryption and decryption happens via the key manager service which is an external api call. If we are encrypting every field of a domain model individually, this would take a lot of network calls and add latency. So we are batch encrypting them.

## How did you test it?

- Check the sanity of payments v2
- Create a payment
- Confirm the payment
- Retrieve the payment

All the above should return 200 status code


## Checklist
